ui/src/app/pages/data/data-set/dataset-page.spec.tsx

```diff
@@ -67,15 +67,78 @@ describe('DataSetPage', () => {
         .find('input').first();
     condition_concept.simulate('change');
     await waitOneTickAndUpdate(wrapper);
-    expect(wrapper.find('[data-test-id="value-list-items"]').length).toBe(2);
+    let valueListItems = wrapper.find('[data-test-id="value-list-items"]');
+    expect(valueListItems.length).toBe(2);
+    let checkedValuesList = valueListItems.filterWhere(value => value.props().checked);
+
+    // All values should be selected by default
+    expect(checkedValuesList.length).toBe(2);
 
     // Second Concept set in concept set list has domain "Measurement"
     const measurement_concept = wrapper.find('[data-test-id="concept-set-list-item"]').at(1)
         .find('input').first();
     measurement_concept.simulate('change');
     await waitOneTickAndUpdate(wrapper);
     await waitOneTickAndUpdate(wrapper);
-    expect(wrapper.find('[data-test-id="value-list-items"]').length).toBe(5);
+    valueListItems = wrapper.find('[data-test-id="value-list-items"]');
+    checkedValuesList = valueListItems.filterWhere(value => value.props().checked);
+    expect(valueListItems.length).toBe(5);
+    expect(checkedValuesList.length).toBe(5);
+  });
+
+  it('should select all values by default on selection on concept set only if the new domain is unique',
+    async() => {
+      const wrapper = mount(<DataSetPage/>);
+      await waitOneTickAndUpdate(wrapper);
+      await waitOneTickAndUpdate(wrapper);
```

Comment:
Is there a reason for the double-waitOneTick call here? I see this pattern is used elsewhere. If it's important to the test, I'd leave a comment to help the reader understand why. Have you tried removing it? If so, what broke?

Reply:
I will confirm this but this is because there are 2 promise calls (one for cohort and another for concept set list)

Reply:
I tested this and one waitOneTick works so have updated the test